finder-app/writer.c

```diff
@@ -0,0 +1,84 @@
+* the user to create a file in a given directory (1st CLI argument) and to write a string or
+* A word that is passed to the executable (2nd CLI argument).
+* The user must enter two arguments in addition to the executable generated by this file, or
+* the file shall return 1, a LOG_ERR severity will be logged, and a write operation on the file will not occur.
+* The executable can be generated for the host(gcc Compiler) and for the target (aarch64 Compiler)
+* more information on how to build this file can be found on the make file
+​​*
+​​*
+​​*​ ​@author​ ​Nileshkartik Ashokkumar
+​​*​ ​@date​ ​Jan​ ​28​ ​2022
+​*​ ​@version​ ​1.0
+​*
+*/
+
+
+/************************************include files***************************/
+#include <syslog.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <libgen.h>
+
+
+/***********************************MACRO Definition*****************************/
+#define VALID_ARGS   	(3)										/*valid number of arguments that is expected while running the program*/
+#define SNV_INT      	(-1)										/*Invalid value notation*/
+#define FILECMD_ARG	(1)										/*1st CLI argument to indicate the file*/
+#define WORDCMD_ARG	(2)										/*2nd CLI argument to indicate the word to be written in the file*/
+
+
+/*********************************Function starts here***************************/
+int main(int argc, char *argv[])
+{
+	int fd_val; 										/*variable to store file descriptor value*/						
+	ssize_t wr_len;										/*Variable to store write bytes length return value*/
+	ssize_t buffer_len;									/*Variable to calculate the string length of the data to be writen 2nd CLI args*/
+	
+	openlog(NULL, 0 , LOG_USER);									/*open connection for sys logging, ident is NULL to use this Program for the user level messages*/
+		
+	if(argc != VALID_ARGS)
+	{
+		syslog(LOG_ERR,"Invlaid number of arguments; current arg count: %d",(argc-1));	/*Syslog with priority error since the user arguments were insufficient or more */
+		return 1;
+	}
+	
+	buffer_len = strlen(argv[WORDCMD_ARG]);							/*calculate the buffer length*/
+	fd_val = open(argv[FILECMD_ARG], O_WRONLY | O_CREAT | O_NONBLOCK, S_IRWXU | S_IRWXG | S_IRWXO);	/*open the file, with Read/ Write/ Exec privileges for user, group and others(umask state is maintained)*/ 
+	
+	if(fd_val != SNV_INT)										
+	{
+		syslog(LOG_DEBUG,"Writing %s to %s",argv[WORDCMD_ARG],basename(argv[FILECMD_ARG]));	/*Debug message to write the word into the file*/
+		wr_len = write(fd_val, argv[WORDCMD_ARG], buffer_len);				/*write the argv[2] of buffer_len in the open file descriptor*/
+		if((wr_len == SNV_INT) || (wr_len != buffer_len))					/*error handling in case of write failed or there is insufficient space in the target file*/
+		{
+			syslog(LOG_ERR,"Write to file failed or insufficient space to write");
+			return 1;
```

Comment:
@Nileshkartik95 missing cleanup `close(fd_val)` in this error path.
Rather than returning here I'd suggest setting a variable you can return at the end of the program, then handling success and failure cases with the same cleanup logic.  See discussion in class today.